app/src/main/java/nl/tudelft/bbw/API.java

```diff
@@ -46,15 +49,34 @@ public static void initializeAPI(User owner, Context context) throws HashExcepti
     /**
      * Method to add a contact to your chain
      *
-     * @param contact the contact
      * @param signature byte array containing the signature
-     * @param message byte array containing the message
+     * @param message   byte array containing the message
      * @throws HashException               When creating a block results in an error
      * @throws BlockAlreadyExistsException when adding a block results in an error
      */
-    public static void addContactToChain(User contact, byte[] signature, byte[] message)
+    public static void addAcquaintanceToChain(Acquaintance acquaintance, byte[] signature, byte[] message)
             throws HashException, BlockAlreadyExistsException {
-        blockController.addBlockToChain(contact, signature, message);
+
+        //Adding his database into your database (so you can look up his contacts)
+        List<List<Block>> multichain = acquaintance.getMultichain();
+        if (multichain.isEmpty()) {
+            return;
+        }
+        for (List<Block> chain : multichain) {
```

Comment:
Try to keep the methods in the API as small as possible, move the actual computing into one of the controllers, that's their function.